apps/ledger-live-mobile/e2e/models/DeviceAction.ts

```diff
@@ -0,0 +1,216 @@
+import { DeviceLike } from "../../src/reducers/types";
+
+const mockListAppsResult = (
+  appDesc: string,
+  installedDesc: string,
+  deviceInfo: DeviceInfo,
+  deviceModelId?: DeviceModelId,
+) => {
+  const result = innerMockListAppResult(appDesc, installedDesc, deviceInfo, deviceModelId);
+  Object.keys(result?.appByName).forEach(key => {
+    result.appByName[key] = { ...result.appByName[key], type: AppType.currency };
+  });
+  return result;
+};
+
+// this implement mock for the "legacy" device action (the one working with live-common/lib/hw/actions/*)
+export default class DeviceAction {
+  deviceLike: DeviceLike;
+  device: Device;
+
+  constructor(deviceLike: DeviceLike) {
+    this.deviceLike = deviceLike;
+    this.device = this.deviceLikeToDevice(deviceLike);
+  }
+
+  deviceLikeToDevice(d: DeviceLike): Device {
+    return {
+      deviceId: d.id,
+      deviceName: d.name,
+      modelId: d.modelId,
+      wired: false,
+    };
+  }
+
+  async selectMockDevice() {
+    const elId = "device-item-" + this.deviceLike.id;
+    await waitForElementById(elId);
+    await tapById(elId);
+  }
+
+  async waitForSpinner() {
+    await waitForElementById("device-action-loading");
+  }
+
+  async openApp() {
+    await this.waitForSpinner();
+    mockDeviceEvent({ type: "opened" });
+  }
+
+  async genuineCheck(appDesc = "Bitcoin", installedDesc = "Bitcoin") {
+    await this.waitForSpinner();
+
+    const { device } = this;
+    const result = mockListAppsResult(appDesc, installedDesc, deviceInfo);
+    mockDeviceEvent(
+      {
+        type: "deviceChange",
+        device,
+      },
+      {
+        type: "listingApps",
+        deviceInfo,
+      },
+      {
+        type: "result",
+        result,
+      },
+      { type: "complete" },
+    );
+  }
+
+  async accessManager(
+    appDesc = "Bitcoin,Tron,Litecoin,Ethereum,Ripple,Stellar",
+    installedDesc = "Bitcoin,Litecoin,Ethereum (outdated)",
+  ) {
+    await this.waitForSpinner();
+
+    const { device } = this;
+
+    const result = mockListAppsResult(appDesc, installedDesc, deviceInfo, device.modelId);
+    mockDeviceEvent(
+      {
+        type: "deviceChange",
+        device,
+      },
+      {
+        type: "listingApps",
+        deviceInfo,
+      },
+      {
+        type: "result",
+        result,
+      },
+      { type: "complete" },
+    );
+  }
+
+  async accessManagerWithL10n(
+    appDesc = "Bitcoin,Tron,Litecoin,Ethereum,Ripple,Stellar",
+    installedDesc = "Bitcoin,Litecoin,Ethereum (outdated)",
+  ) {
+    await this.waitForSpinner();
+
+    const { device } = this;
+
+    const result = mockListAppsResult(appDesc, installedDesc, deviceInfo210lo5);
+    mockDeviceEvent(
+      {
+        type: "deviceChange",
+        device,
+      },
+      {
+        type: "listingApps",
+        deviceInfo: deviceInfo210lo5,
+      },
+      {
+        type: "result",
+        result,
+      },
+      { type: "complete" },
+    );
+  }
+
+  async complete() {
```

Comment:
I don't think you need `async` here

Reply:
from the external perspective (user of the model) it's best to keep these returning a promise so the consumer is going to have to still do `await` because we may actually need to add some async stuff when we start using it (some of these functions aren't used yet)